`test/integration/uv` failing when run in a container

We're trying to update the version of raft in Debian and are encountering issues with the tests failing in containerized environments, but not on physical hardware or full VMs. If I build raft in a LXD sid container running on a bookworm host, I get several test failures running `test/integration/uv` (duplicates omitted for brevity):

```
recv/first                                                  [ ERROR ]
Error: test/integration/test_uv_recv.c:256: assertion failed: rv == 0 (18 == 0)
Error: child killed by signal 6 (Aborted)

Error: test/integration/test_uv_send.c:141: uv_run: condition not met in 20 iterations
Error: child killed by signal 6 (Aborted)

tcp_connect/first                                           [ ERROR ]
Error: test/integration/test_uv_tcp_connect.c:47: assertion failed: status == result->status (16 == 0)
Error: child killed by signal 6 (Aborted)

tcp_connect/refused                                         [ ERROR ]
Error: test/integration/test_uv_tcp_connect.c:222: assertion failed: string f->transport.errmsg == "uv_tcp_connect(): connection refused" ("uv_getaddrinfo(): EAI_NONAME" == "uv_tcp_connect(): connection refused")
Error: child killed by signal 6 (Aborted)
```

And then the test seems to hang forever:

```
tcp_connect/closeDuringDnsLookupAbort                       [ OK    ] [ 0.00051087 / 0.00069748 CPU ]
tcp_connect/closeDuringConnectAbort                         ^C
```

If I build raft on my physical bookworm system or within a QEMU VM running sid, the tests all pass.

Laszlo also reports encountering the following error when attempting to build within a chroot:

```
Error: test/unit/test_uv_fs.c:320: assertion failed: rv_ == RAFT_IOERR (0 == 18)
```

I see this behavior with both raft 0.17.1 and current master. Since it seems to involve libuv, that library is currently at version 1.44.2 in both bookworm and sid. There may be some overlap with canonical/dqlite#581.

I tried to update dqlite's packaging in Debian to v1.16.4, but am encountering this same issue when trying to build with the now-bundled raft source. `tcp_connect/closeDuringConnectAbort` still hangs, and there are several additional test failures (due to output size, I'm not including the various failed tests):

```
180 of 233 (77%) tests successful, 11 (5%) test skipped.
FAIL raft-uv-integration-test (exit status: 1)
```

Trying to build with the current cowsql/raft fork results in an error:

```
libtool: link: gcc -std=c11 -g3 -fcf-protection --param=ssp-buffer-size=4 -pipe -fno-strict-aliasing -fdiagnostics-color -fexceptions -fstack-clash-protection -fstack-protector-strong -fasynchronous-unwind-tables -fdiagnostics-show-option -Wall -Wextra -Wimplicit-fallthrough=5 -Wcast-align -Wstrict-prototypes -Wlogical-op -Wmissing-include-dirs -Wold-style-definition -Winit-self -Wfloat-equal -Wsuggest-attribute=noreturn -Wformat=2 -Wshadow -Wendif-labels -Wdate-time -Wnested-externs -Wconversion -Werror -O2 -Wno-conversion -g -O2 -ffile-prefix-map=/build/dqlite-1.16.4=. -fstack-protector-strong -fstack-clash-protection -Wformat -Werror=format-security -fcf-protection -Wl,-z -Wl,relro -Wl,-z -Wl,now -o integration-test test/integration/integration_test-test_client.o test/integration/integration_test-test_cluster.o test/integration/integration_test-test_fsm.o test/integration/integration_test-test_membership.o test/integration/integration_test-test_node.o test/integration/integration_test-test_role_management.o test/integration/integration_test-test_server.o test/integration/integration_test-test_vfs.o test/integration/integration_test-main.o  ./.libs/libtest.a -luv -ldl -lrt -lpthread -lsqlite3 -lraft ./.libs/libdqlite.so -pthread -Wl,-rpath -Wl,/build/dqlite-1.16.4/.libs
/usr/bin/ld: ./.libs/libdqlite.so: undefined reference to `raft_register_state_cb'
collect2: error: ld returned 1 exit status
```

Possibly relevant versions: Debian sid LXD container with Debian kernel 6.1.0-18-amd64; libsqlite 3.45.1; libuv 1.48.0; liblz4 1.9.4.

> I tried to update dqlite's packaging in Debian to v1.16.4, but am encountering this same issue when trying to build with the now-bundled raft source. `tcp_connect/closeDuringConnectAbort` still hangs, and there are several additional test failures (due to output size, I'm not including the various failed tests):
> 
> ```
> 180 of 233 (77%) tests successful, 11 (5%) test skipped.
> FAIL raft-uv-integration-test (exit status: 1)
> ```

This could be fixed by applying a change like in https://github.com/cowsql/raft/pull/79, see:

https://github.com/cowsql/raft/pull/79/commits/1a00ebf0482000d8b62ea1932bb88122a951aaba

in particular.

> Trying to build with the current cowsql/raft fork results in an error:
> 
> ```
> libtool: link: gcc -std=c11 -g3 -fcf-protection --param=ssp-buffer-size=4 -pipe -fno-strict-aliasing -fdiagnostics-color -fexceptions -fstack-clash-protection -fstack-protector-strong -fasynchronous-unwind-tables -fdiagnostics-show-option -Wall -Wextra -Wimplicit-fallthrough=5 -Wcast-align -Wstrict-prototypes -Wlogical-op -Wmissing-include-dirs -Wold-style-definition -Winit-self -Wfloat-equal -Wsuggest-attribute=noreturn -Wformat=2 -Wshadow -Wendif-labels -Wdate-time -Wnested-externs -Wconversion -Werror -O2 -Wno-conversion -g -O2 -ffile-prefix-map=/build/dqlite-1.16.4=. -fstack-protector-strong -fstack-clash-protection -Wformat -Werror=format-security -fcf-protection -Wl,-z -Wl,relro -Wl,-z -Wl,now -o integration-test test/integration/integration_test-test_client.o test/integration/integration_test-test_cluster.o test/integration/integration_test-test_fsm.o test/integration/integration_test-test_membership.o test/integration/integration_test-test_node.o test/integration/integration_test-test_role_management.o test/integration/integration_test-test_server.o test/integration/integration_test-test_vfs.o test/integration/integration_test-main.o  ./.libs/libtest.a -luv -ldl -lrt -lpthread -lsqlite3 -lraft ./.libs/libdqlite.so -pthread -Wl,-rpath -Wl,/build/dqlite-1.16.4/.libs
> /usr/bin/ld: ./.libs/libdqlite.so: undefined reference to `raft_register_state_cb'
> collect2: error: ld returned 1 exit status
> ```

This is a known issue, and wouldn't be hard to fix, but since dqlite is now shipping with an embedded raft source code it's probably not worth it, unless somebody really needs that.

--- a/src/raft/uv_ip.c
+++ b/src/raft/uv_ip.c
@@ -58,7 +58,7 @@ int uvIpAddrSplit(const char *address,
 int uvIpResolveBindAddresses(const char *address, struct addrinfo **ai_result)
 {
 	static struct addrinfo hints = {
-	    .ai_flags = AI_ADDRCONFIG | AI_PASSIVE | AI_NUMERICSERV,
+	    .ai_flags = AI_PASSIVE | AI_NUMERICSERV,
 	    .ai_family = AF_INET,
 	    .ai_socktype = SOCK_STREAM,
 	    .ai_protocol = 0};

--- a/src/raft/uv_tcp_connect.c
+++ b/src/raft/uv_tcp_connect.c
@@ -261,7 +261,7 @@ static void uvGetAddrInfoCb(uv_getaddrinfo_t *req,
 /* Create a new TCP handle and submit a connection request to the event loop. */
 static int uvTcpConnectStart(struct uvTcpConnect *r, const char *address)
 {
-	static struct addrinfo hints = {.ai_flags = AI_V4MAPPED | AI_ADDRCONFIG,
+	static struct addrinfo hints = {.ai_flags = 0,
 					.ai_family = AF_INET,
 					.ai_socktype = SOCK_STREAM,
 					.ai_protocol = 0};

